src/julia_gc.c

```diff
@@ -597,8 +607,31 @@ static void SetupSafeRestoreHandlers(void)
     set_safe_restore = set_safe_restore_with_offset;
 }
 
+#else // SKIP_GUARD_PAGES
+
+static size_t guardpages_size;
+
+void SetupGuardPagesSize(void) {
+    // This is a generic implementation that assumes that all threads
+    // have the default guard pages. This should be correct for the
+    // current Julia implementation.
+    pthread_attr_t attr;
+    pthread_attr_init(&attr);
+    if (pthread_attr_getguardsize(&attr, &guardpages_size) < 0) {
```

Comment:
unfortunately this causes a linker error in the CI:
```
undefined reference to symbol 'pthread_attr_getguardsize@@GLIBC_2.2.5'
```
I guess we need to pass a suitable linker flag, something like `-pthreads` or so?

Reply:
Ah yes, that's a Linux thing where you need to use `-pthread` to compile and link. I think this is something that I should add to the configure script for the `--with-gc=julia` case (Julia already requires pthreads, so it's just a matter of getting compilation and linking right).